unit-tests/src/test/java/org/eclipse/collections/impl/map/immutable/ImmutableMapIterableTestCase.java

```diff
@@ -383,6 +383,81 @@ public void newWithKeyValue()
         Verify.assertSize(immutable.size() + 1, immutable2);
     }
 
+    @Test
+    public void newWithMap1()
+    {
+        ImmutableMapIterable<Integer, String> immutable = this.classUnderTest();
+        ImmutableMapIterable<Integer, String> immutable2 =
+                immutable.newWithMap(UnifiedMap.newMapWith(
+                        Tuples.pair(
+                                Integer.MAX_VALUE,
+                                Integer.toString(Integer.MAX_VALUE)),
+                        Tuples.pair(
+                                Integer.MIN_VALUE,
+                                Integer.toString(Integer.MIN_VALUE))));
+        Verify.assertSize(immutable.size() + 2, immutable2);
+    }
+
+    @Test
+    public void withMapNull()
+    {
+        Assert.assertThrows(NullPointerException.class, () -> this.classUnderTest().newWithMap(null));
+    }
+
+    @Test
+    public void newWithMapTargetEmpty()
+    {
+        ImmutableMapIterable<Integer, String> immutable = this.classUnderTest();
+        ImmutableMapIterable<Integer, String> immutable2 = immutable.newWithMap(UnifiedMap.newMap());
+        Assert.assertNotSame(immutable, immutable2);
+        Verify.assertMapsEqual(immutable2.castToMap(), immutable.castToMap());
+    }
+
+    @Test
+    public void newWithMapEmptyAndTargetEmpty()
+    {
+        ImmutableMapIterable<Integer, String> immutable = this.classUnderTest();
+        ImmutableMapIterable<Integer, String> immutable2 = immutable.newWithMap(Maps.mutable.empty());
+        Assert.assertNotSame(immutable, immutable2);
+        Verify.assertMapsEqual(immutable2.castToMap(), immutable.castToMap());
+    }
+
+    @Test
+    public void newWithMapIterable()
+    {
+        ImmutableMapIterable<Integer, String> immutable = this.classUnderTest();
+        ImmutableMapIterable<Integer, String> immutable2 =
+                immutable.newWithMapIterable(Maps.immutable.of(
+                        Integer.MAX_VALUE,
+                        Integer.toString(Integer.MAX_VALUE),
+                        Integer.MIN_VALUE,
+                        Integer.toString(Integer.MIN_VALUE)));
+        Verify.assertSize(immutable.size() + 2, immutable2);
```

Comment:
I think we need better assertions here. It should be possible to assert with a `MutableMap` no?

Reply:
Updated the code to handle assert with a MutableMap using Verify.assertMapsEqual() method.